### DISCUSSION TITLE
Which Search Engines do you use and why?

### DISCUSSION THREAD

**Discussion by alexander-schranz** [schranz-search/schranz-search#457] [General]
There are a lot of different search engines. In SEAL we are already supporting Elasticsearch, Opensearch, Algolia, Meilisearch, Redisearch, Typesense, Solr and Loupe.

I want to know which Search Engines do you use and why? Even if you are not using SEAL or a supported search engine of SEAL.

So my questions are:

 - **What Search Engines do you use?**
 - **Which features do you use of your Search Engines?**
 - **How did you decide to use it?**

**Comment by Ocramius** [comment]
> What Search Engines do you use?

PostgreSQL, ManticoreSearch.

> Which features do you use of your Search Engines?

Fuzzy search, multi-language support, aggregation, faceting, relational operations

> How did you decide to use it?

PostgreSQL is old/stable and extremely good for non-aggregation operations. Hosted solutions exist everywhere.

ManticoreSearch kinda "happened", but is extremely fast at aggregations, although a very wonky engine with its own quirks that one needs to learn to live with.

**Reply by alexander-schranz** [reply]
Thx for your response. What are typical aggregation / facetings you have todo? Aggregation sure a topic which is on my list also for SEAL.

ManticoreSearch was something I also look into it but did sadly not yet get it work in all cases the way I expected it to fulfill yet [when I tried it](https://github.com/schranz-search/schranz-search/pull/103).

**Reply by Ocramius** [reply]
The usual scenario is:

 * "find possible prices" or "find colours"
 * but only for customer x
 * and only for user group y
 * and only when in category z or sub-categories

This is roughly e-commerce stuff with more detailed access control.

**Comment by eislambey** [comment]
> What Search Engines do you use?

Typesense

> Which features do you use of your Search Engines?

- Search as you type
- Curations
- Geo search
- Synonyms

> How did you decide to use it?

it's easy to implement, has good demos on their website and provides hosted service.

**Reply by alexander-schranz** [reply]
Curations / Overrides looks like a nice feature of typesense thanks for your response here :)

**Comment by Crovitche-1623** [comment]
Meilisearch. Handy for supporting typos and for its speed.

**Reply by alexander-schranz** [reply]
They have a really nice SDK/API very friendly for even people not very deep in the search topic.

**Comment by steeven-th** [comment]
What Search Engines do you use?
Algolia

Which features do you use of your Search Engines?
Multi-languages
Typo-tolerance
Synonyms

How did you decide to use it?
Free usage for 10k search requests / month

**Reply by alexander-schranz** [reply]
Do you use replicas in Algolia for example for different sortings?

**Reply by steeven-th** [reply]
The situation hasn’t arisen, but if we had to, yes.

**Comment by lolaslade** [comment]
What Search Engines do you use?
Solr, Meilisearch, Elasticsearch on different projects. Planning to test Typesense

Which features do you use of your Search Engines?
Multi-languages, faceting, synonyms, geo search

How did you decide to use it?
Solr - OOTB Drupal support
Meilisearch - Easy deployment and maintenance
Elasticsearch - Great tools for developers to debug search

**Comment by jbfewo** [comment]
We set up a blank Sulu 3 that uses SEAL for website and admin area search. The loupe adapter works effortlessly for local development because it is configured as default after install (after figuring out, that you have to create a search index once manually).

In preparation for the production deployment I wanted to switch to an adapter that could be used by multiple instances, as our app may scale vertically and a local per-pod index would be a bad design. As we already use a shared Redis for sessions and caching, Redisearch was an obvious choice. THX to @alexander-schranz Redisearch and Sulu3 are working great together since today!

So the main reason for Redisearch was, that it fits seamlessly into the existing tech stack.

**Reply by alexander-schranz** [reply]
Thx for letting us know. That sounds great, keep in mind that you may want maybe 2 different instances of Redis for One for Cache/Sessions and One for Search. As cache/sessions mostly run on some cache policy like `allkeys-lru` and `max-memory` limits and you don't want that it starts dropping your search documents.

**Reply by jbfewo** [reply]
Thats right, and thats even the way redis recommends¹ such cases. But currently I postpone that topic to when site size and user count reach a relevant level to hit `maxmemory`... and circumvent it by setting `maxmemory-policy volatile-ttl` 

[¹](https://redis.io/docs/latest/develop/reference/eviction/#eviction-policies)
